lib/v2/screens/explore_screens/invite/interactor/viewmodels/invite_bloc.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:bloc/bloc.dart';
+import 'package:seeds/v2/blocs/rates/viewmodels/rates_state.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/screens/explore_screens/invite/interactor/viewmodels/bloc.dart';
+
+import '../../../../../datasource/remote/model/balance_model.dart';
+import 'package:seeds/v2/utils/rate_states_extensions.dart';
+
+/// --- BLOC
+class InviteBloc extends Bloc<InviteEvent, InviteState> {
+  InviteBloc(RatesState rates) : super(InviteState.initial(rates));
```

Comment:
I don't think bloc should rely on other states, I would pass what's needed from RateState in the event instead.

Reply:
Yes I already know in fact past throught a event data is how we usually do it.

Now I not consider this a bad practice passing a dependency to the block, this is usually when you need to have a data in the init state long before it is painted and mounted the widget (There are few scenaries but happens).
 
To be honest I agree with you here I think these extencion funtions must be reside on the double or String clases instead. To prevent this dependency on RateState. Something like: 

`24.fromSeedsToFiatString('USD');`

@gguijarro-c-chwy What do you thing about this approach for conversion extencions?

Reply:
Ok I just saw because the extension is on `RateState` because `RateModel` and `FiatRateModel` have the rate seeds and the rates currencies.